typing_extensions/src_py3/typing_extensions.py

```diff
@@ -503,6 +504,142 @@ class Other(Leaf):  # Error reported by type checker
     return f
 
 
+if hasattr(typing, 'Literal'):
+    Literal = typing.Literal 
+elif sys.version_info[:2] >= (3, 7):
+    class _LiteralForm(typing._SpecialForm, _root=True):
+
+        def __repr__(self):
+            return 'typing_extensions.' + self._name
+
+        def __getitem__(self, parameters):
+            return _GenericAlias(self, parameters)
+
+    Literal = _LiteralForm('Literal', doc=
+        """A type that can be used to indicate to type checkers that the
+        corresponding value has a value literally equivalent to the
+        provided parameter. For example:
+
+            var: Literal[4] = 4
+        
+        The type checker understands that 'var' is literally equal to the
+        value 4 and no other value.
+
+        Literal[..] cannot be subclassed. There is no runtime checking
```

Comment:
Are two dots (not three) intentional? It is a bit unusual.

Reply:
Whoops, that was an oversight -- fixed!